demos/mqtt/mqtt_demo_lightweight/mqtt_config.h

```diff
@@ -45,9 +45,19 @@
 #endif
 
 #include "logging_stack.h"
+
 /************ End of logging configuration ****************/
 
-/* Set network context to socket (int). */
-typedef int NetworkContext_t;
+/**
+ * @brief The maximum number of MQTT PUBLISH messages that may be pending
+ * acknowledgement at any time.
+ *
+ * QoS 1 and 2 MQTT PUBLISHes require acknowledgement from the server before
+ * they can be completed. While they are awaiting the acknowledgement, the
+ * client must maintain information about their state. The value of this
+ * macro sets the limit on how many simultaneous PUBLISH states an MQTT
+ * context maintains.
+ */
+#define MQTT_STATE_ARRAY_MAX_COUNT    10U
```

Comment:
Is this needed for the lightweight demo? It doesn't use an `MQTTContext_t` so this shouldn't need to be defined